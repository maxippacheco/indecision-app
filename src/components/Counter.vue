<template>
	<h2>{{ customTitle }}</h2>
	<p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
	<p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
	<p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
	<h3>{{ counter }}</h3>
	<div>
		<button @click="increase">+1</button>
		<button @click="decrease">-1</button>
	</div>
</template>

<script>
export default {
	// props: ['title', 'start'],
	props: {
		title: String,
		start:{
			type: Number,
			required: true,
			validator(value){
				return value >= 0
			}
		} 
	},
	name: 'Counter',
	data(){
		return {
			counter: this.start
		}	
	},
	methods: {
		getSquare(){
			return this.counter * this.counter;
		},
		increase(){
			this.counter++;
		},
		decrease(){
			this.counter--;
		}
	},
	// guarda las propiedades en el cache
	computed: {
		squareCounter(){
			console.log('Computed squareCounter');
			return this.counter * this.counter;
		},
		customTitle(){
			return this.title ? this.title : 'Counter'
		}
	}
}
</script>

<style>

button{
	background-color: #64BB87;
	border-radius: 5px;
	color: white;
	border: none;
	padding: 10px;
	margin: 5px 15px;
	cursor: pointer;
	transition: 0.3s ease-in-out;
}

button:hover{
	background-color: #478761;
	transition: 0.3s ease-in-out;
}

</style>