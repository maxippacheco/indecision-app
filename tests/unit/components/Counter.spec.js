import { mount, shallowMount } from "@vue/test-utils";
import Counter from '@/components/Counter'

describe('Pruebas en Counter', () => {
	// test('Debe de hacer match con el snapshot', () => {

	// 	const wrapper = shallowMount( Counter )

	// 	expect( wrapper.html() ).toMatchSnapshot();
	// })

	test('Should render h2 w default value', () => {

		const wrapper = shallowMount( Counter )
	
		const h2 = wrapper.find('h2');

		expect( h2.text() ).toBe('Counter')
	
	})
})