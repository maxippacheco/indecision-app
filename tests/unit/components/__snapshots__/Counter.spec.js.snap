// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Counter Debe de hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p> <sup>2</sup> = NaN</p>
<p> <sup>2</sup> = NaN</p>
<p> <sup>2</sup> = NaN</p>
<h3></h3>
<div><button>+1</button><button>-1</button></div>
`;
